docs/quickstart.md

```diff
@@ -2,198 +2,441 @@
+
+You can specify the regularization scheme and its strength when initializing the GLM model, as shown below:
+
+
+```python
+
+>>> import nemos as nmo
+
+>>> # Instantiate a GLM with Ridge (L2) regularization
+>>> glm = nmo.glm.GLM(regularizer="Ridge", regularizer_strength=0.1)
+
+```
+
+
+
+## __Pre-processing with `pynapple`__
 
-### Pre-processing with `pynapple`
 
 !!! warning
-    This section assumes some familiarity with the `pynapple` package for time series manipulation and data 
+
+    This section assumes some familiarity with the `pynapple` package for time series manipulation and data
     exploration. If you'd like to learn more about it, take a look at the [`pynapple` documentation](https://pynapple-org.github.io/pynapple/).
 
-`pynapple` is an extremely helpful tool when working with time series data. You can easily perform operations such 
-as restricting your time series to specific epochs (sleep/wake, context A vs. context B, etc.), as well as common 
+
+`pynapple` is an extremely helpful tool when working with time series data. You can easily perform operations such
+as restricting your time series to specific epochs (sleep/wake, context A vs. context B, etc.), as well as common
 pre-processing steps in a robust and efficient manner. This includes bin-averaging, counting, convolving, smoothing and many
 others. All these operations can be easily concatenated for a quick and easy data pre-processing.
 
-In NeMoS, if a transformation  preserve the time axis and you use a `pynapple` time series as input, the result will 
+In NeMoS, if a transformation  preserve the time axis and you use a `pynapple` time series as input, the result will
 also be a `pynapple` time series.
 
 A canonical example of this behavior is the `predict` method of `GLM`.
 
+
 ```python
->>>  # Assume X is a pynapple TsdFrame
+
+>>> import numpy as np
+>>> import pynapple as nap
+
+>>> # create a TsdFrame with the features and a Tsd with the counts
+>>> X = nap.TsdFrame(t=np.arange(100), d=np.random.normal(size=(100, 2)))
+>>> y = nap.Tsd(t=np.arange(100), d=np.random.poisson(size=(100, )))
+
 >>> print(type(X))  # shape (num samples, num features)
 <class 'pynapple.core.time_series.TsdFrame'>
 
->>> model.fit(X, y)  # the following works
+>>> model = model.fit(X, y)  # the following works
 
 >>> firing_rate = model.predict(X)  # predict the firing rate of the neuron
 
 >>>  # this will still be a pynapple time series
 >>> print(type(firing_rate))  # shape (num_samples, )
 <class 'pynapple.core.time_series.Tsd'>
+
 ```
 
 Let's see how you can greatly streamline your analysis pipeline by integrating `pynapple` and NeMoS.
```

Comment:
do we want to include this whole example? or just link to the full example in the docs?

I think, regardless, we should link to the example for more details

Reply:
I would live this, since it shows really how good the tools work together.